x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/__jest__/test_pipeline.test.tsx

```diff
@@ -0,0 +1,241 @@
+      expect(find('outputTab').props().disabled).toEqual(false);
+      expect(exists('documentsTabContent')).toBe(false);
+      expect(exists('outputTabContent')).toBe(true);
+
+      // Click reload button and verify request
+      const totalRequests = server.requests.length;
+      await actions.clickRefreshOutputButton();
+      expect(server.requests.length).toBe(totalRequests + 1);
+      expect(server.requests[server.requests.length - 1].url).toBe(
+        '/api/ingest_pipelines/simulate'
+      );
+
+      // Click verbose toggle and verify request
+      await actions.toggleVerboseSwitch();
+      expect(server.requests.length).toBe(totalRequests + 2);
+      expect(server.requests[server.requests.length - 1].url).toBe(
+        '/api/ingest_pipelines/simulate'
+      );
+    });
+
+    test('should enable the output tab if cached documents exist', async () => {
+      const { actions, exists } = testBed;
+
+      httpRequestsMockHelpers.setSimulatePipelineResponse(SIMULATE_RESPONSE);
+
+      // Open flyout
+      actions.clickAddDocumentsButton();
+
+      // Add sample documents and click run
+      actions.addDocumentsJson(JSON.stringify(DOCUMENTS));
+      await actions.clickRunPipelineButton();
+
+      // Close flyout
+      actions.closeTestPipelineFlyout();
+      expect(exists('testPipelineFlyout')).toBe(false);
+      expect(exists('addDocumentsButton')).toBe(false);
+      expect(exists('documentsDropdown')).toBe(true);
+
+      // Reopen flyout and verify output tab is enabled
+      await actions.clickViewOutputButton();
+      expect(exists('testPipelineFlyout')).toBe(true);
+      expect(exists('documentsTabContent')).toBe(false);
+      expect(exists('outputTabContent')).toBe(true);
+    });
+
+    test('should surface API errors from the request', async () => {
+      const { actions, find, exists } = testBed;
+
+      const error = {
+        status: 400,
+        error: 'Bad Request',
+        message:
+          '"[parse_exception] [_source] required property is missing, with { property_name="_source" }"',
+      };
+
+      httpRequestsMockHelpers.setSimulatePipelineResponse(undefined, { body: error });
+
+      // Open flyout
+      actions.clickAddDocumentsButton();
+
+      // Add invalid sample documents array and run the pipeline
+      actions.addDocumentsJson(JSON.stringify([{}]));
+      await actions.clickRunPipelineButton();
+
+      // Verify error rendered
+      expect(exists('pipelineExecutionError')).toBe(true);
+      expect(find('pipelineExecutionError').text()).toContain(error.message);
+    });
+  });
+
+  describe('Processors', () => {
+    // This is a hack
+    // We need to provide the processor id in the mocked output;
+    // this is generated dynamically and not something we can stub.
+    // As a workaround, the value is added as a data attribute in the UI
+    // and we retrieve it to generate the mocked output.
+    const addProcessorTagtoMockOutput = (output: VerboseTestOutput) => {
```

Comment:
I guess that another way would have been to mock the `v4()` method of the `uuid` package and return a sequential id. So if there are 5 processors, their id would be 0, 1, 2, 3, 4.

Reply:
Good point. I'm going to leave this as-is for now, but will consider refactoring it as I add more tests.